python/ray/data/tests/test_progress_bar.py

```diff
@@ -66,3 +66,15 @@ def wrapped_close():
 
     assert pb._progress == new_total
     assert total_at_close == new_total
+
+    # Test updating the total
+    pb = ProgressBar("", total, enabled=True)
+    assert pb._bar is not None
+    patch_close(pb._bar)
+    new_total = total * 2
+    pb.update(0, new_total)
+
+    assert pb._bar.total == new_total
+    pb.update(total + 1, total)
+    assert pb._bar.total == total + 1
+    pb.close()
```

Comment:
Let's also add an operator-level test to make sure the block estimation is correct. The test logic could be something like  this:
* define a map op with an UDF that will yield N blocks. 
* manually add input blocks to the op one by one (via `add_input`). 
* Check the estimated block number and the progress bar total is correct. 

This test can be added to python/ray/data/tests/test_operators.py